codegen/module_system.go

```diff
@@ -1054,6 +1076,80 @@ func (generator *GatewayServiceGenerator) Generate(
 	}, nil
 }
 
+/*
+ * Middleware Generator
+ */
+
+// MiddlewareGenerator generates a group of zanzibar endpoints that proxy corresponding clients
+type MiddlewareGenerator struct {
+	templates     *Template
+	packageHelper *PackageHelper
+}
+
+// Generate returns the endpoint build result, which contains a file per
+// endpoint handler and a list of handler specs
+func (g *MiddlewareGenerator) Generate(
+	instance *ModuleInstance,
+) (*BuildResult, error) {
+	ret := map[string][]byte{}
+
+	//	clientSpecs := readClientDependencySpecs(instance)
+
+	dependencies, err := GenerateDependencyStruct(
+		instance,
+		g.packageHelper,
+		g.templates,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"Error generating service dependencies for %s %s",
+			instance.InstanceName,
+			instance.ClassName,
+		)
+	}
+	if dependencies != nil {
+		ret["module/dependencies.go"] = dependencies
+	}
+
+	err = g.generateMiddlewareFile(instance, ret)
+
+	if err != nil {
+		return nil, errors.Wrapf(
+			err,
+			"Error generating middleware file for %s %s",
+			instance.InstanceName,
+			instance.ClassName,
+		)
+	}
+	return &BuildResult{
+		Files: ret,
+		Spec:  nil,
+	}, nil
+}
+
+func (g *MiddlewareGenerator) generateMiddlewareFile(instance *ModuleInstance, out map[string][]byte) error {
+
+	//meta := &MiddlewareMeta{
```

Comment:
is these comments needed?

Reply:
had uncommited changes to remove them, doh!